gtsam/sfm/SfmTrack.h

```diff
@@ -0,0 +1,133 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file SfmTrack.h
+ * @date January 2022
+ * @author Frank Dellaert
+ * @brief A simple data structure for a track in Structure from Motion
+ */
+
+#pragma once
+
+#include <gtsam/base/serialization.h>
+#include <gtsam/geometry/Point2.h>
+#include <gtsam/geometry/Point3.h>
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace gtsam {
+
+/// A measurement with its camera index
+typedef std::pair<size_t, Point2> SfmMeasurement;
+
+/// Sift index for SfmTrack
+typedef std::pair<size_t, size_t> SiftIndex;
+
+/**
+ * @brief An SfmTrack stores SfM measurements grouped in a track
+ * @addtogroup sfm
+ */
+struct SfmTrack {
+  Point3 p;       ///< 3D position of the point
+  float r, g, b;  ///< RGB color of the 3D point
+
+  /// The 2D image projections (id,(u,v))
+  std::vector<SfmMeasurement> measurements;
+
+  /// The feature descriptors
+  std::vector<SiftIndex> siftIndices;
+
+  /// @name Constructors
+  /// @{
+
+  explicit SfmTrack(float r = 0, float g = 0, float b = 0)
+      : p(0, 0, 0), r(r), g(g), b(b) {}
+
+  explicit SfmTrack(const gtsam::Point3& pt, float r = 0, float g = 0,
+                    float b = 0)
+      : p(pt), r(r), g(g), b(b) {}
+
+  /// @}
+  /// @name Standard Interface
+  /// @{
+
+  /// Add measurement (camera_idx, Point2) to track
+  void addMeasurement(size_t idx, const gtsam::Point2& m) {
```

Comment:
this will be a breaking change too for GTSFM, which may be ok, but just something to be aware of (we are using `add_measurement()` currently.

Reply:
I'm aware